libs/auth/src/certs.ts

```diff
@@ -370,20 +393,39 @@ export function constructCardCertSubjectWithoutJurisdictionAndCardType(): string
   return certSubject;
 }
 
+/**
+ * Parses the provided cert and returns machine details. Throws an error if the cert doesn't follow
+ * VotingWorks's machine cert format.
+ */
+export async function parseMachineDetailsFromCert(
+  cert: Buffer
+): Promise<MachineDetails> {
+  const certDetails = await parseCert(cert);
+  assert(certDetails.component !== 'card', 'Cert is not a machine cert');
+  const { component, machineId } = certDetails;
+  return {
+    machineType: component,
+    machineId,
+    jurisdiction: component === 'admin' ? certDetails.jurisdiction : undefined,
+  };
+}
+
 /**
  * Constructs a VotingWorks machine cert subject that can be passed to an openssl command
  */
-export function constructMachineCertSubject(
-  machineType: MachineType,
-  jurisdiction?: string
-): string {
+export function constructMachineCertSubject({
+  machineType,
+  machineId,
+  jurisdiction,
+}: MachineDetails): string {
   assert(
```

Comment:
we may want to assert a format of `machineId` so we don't get in trouble later if it's got weird characters.

Reply:
Along these lines, `machineId` could then be typed `MachineId`, a "newtype"'d `string`.

Reply:
We enforce the structure of the machine ID in the vxsuite-complete-system script that sets it, so I'm going to continue to rely on that for now: https://github.com/votingworks/vxsuite-complete-system/blob/78f0858f525def8bfc0c08d84a77a4808649cc7b/config/vendor-functions/choose-vx-machine-id.sh#L13-L23